### DISCUSSION TITLE
Allow to define in Component which view file to use

### DISCUSSION THREAD

**Discussion by mitchdesign** [livewire/livewire#4794] [1. Help]
Below is a short explanation of my use case, and then the question/feature idea.

### Use case

In my current project I find myself creating similar components that each need a few properties to get the right behavior. They are Filters that will allow users to filter different types of content. So I have for instance:
```
\App\Http\Livewire\Filter\AbstractFilter - which contains the bulk of the shared functionality
\App\Http\Livewire\Filter\Companiesfilter extends AbstractFilter - which contains some specifics about how to filter a list of companies
```
And then in the 'companies' screen I include my component with
```
<livewire:filters.companiesfilter>
```

But, by default this component will look for a Blade view `/resources/views/livewire/filters/companiesfilter.blade.php`. And another subclass will look for another file. While in fact they all need the same view. The differences in the components are for backend logic, not for differences in the front end view. They all should use the same one.

So currently I implemented a render() method in the AbstractFilter 'base' component. But that seems like a bit of an anti pattern because I do not know exactly what magic is happening when Livewire renders views. At the least, I need to pass $_instance into the view or it is not going to work.

### Question

Is there an official way to specify which view a Component should use to render?

### Feature idea

If there is no such way, I think it would be useful to optionally implement something like `getViewName()` or a `protected $view` in the component to allow a more official way to have non-standard component->view relationships.

I would start making a PR but wanted to check first, is there already something in place that I do not know about?

**Comment by joshhanley** [comment]
@mitchdesign render in the parent component should work fine, why is that an anti-pattern?

To use different views in a Livewire component you should use the render method. https://laravel-livewire.com/docs/2.x/rendering-components#render-method

Hope this helps!

**Reply by mitchdesign** [reply]
@joshhanley I was writing `view(...)->render()` in the render method. And then found I had to add `with('_instance' => $this)` to make that work. That felt like something I shouldn't have to do, hence the anti pattern :)

Returning a view without calling render() makes much more sense indeed. Livewire taking care of the plumbing.  That was pretty much the same thing I was asking for. 

Thanks for the hint!

**Reply by joshhanley** [reply]
@mitchdesign ah right! No worries 🙂
